src/server/session.ts

```diff
@@ -1245,7 +1245,7 @@ namespace ts.server {
             // if specified a project, we only return affected file list in this project
             const projectsToSearch = args.projectFileName ? [this.projectService.findProject(args.projectFileName)] : info.containingProjects;
             for (const project of projectsToSearch) {
-                if (project.compileOnSaveEnabled && project.languageServiceEnabled) {
+                if (project.compileOnSaveEnabled && project.languageServiceEnabled && !project.getCompilerOptions().noEmit) {

```

Comment:
Can `noEmit` be undefined?  Would it be safer to check for `true` explicitly?

Reply:
This should be safe because the default value of `noEmit` is false.